File: 771. Jewels and Stones.py
J = "aA"
S = "aAAbbbb"
jewel = [str(i) for i in str(J)]
stone = [str(i) for i in str(S)]
total = 0
for i in stone:
    for j in jewel:
        if i == j:
            total += 1
print(total)

File: Javascript/771. Jewels and Stones.js
var numJewelsInStones = function (jewels, stones) {
    const jewelsArr = jewels.split("");
    const hash = {};
    for (let i = 0; i < stones.length; i++) {
        if (!hash[stones[i]]) {
            hash[stones[i]] = 0;
        }
        hash[stones[i]] += 1;
    }

    let jewel = 0;
    jewelsArr.map((val) => {
        jewel += hash[val] ?? 0;
    });

    return jewel;
};

numJewelsInStones((jewels = "aA"), (stones = "aAAbbbb"));
